numba/core/ir_utils.py

```diff
@@ -593,6 +593,8 @@ def remove_dead_block(block, lives, call_table, arg_aliases, alias_map,
             stmt = f(stmt, lives, lives_n_aliases, arg_aliases, alias_map, func_ir,
                      typemap)
             if stmt is None:
+                if config.DEBUG_ARRAY_OPT >= 2:
+                    print("Statement was removed.")
```

Comment:
Perhaps all these ought to also print the statement that was removed?

Reply:
Each statement is already printed as it is being processed on line 582.  So, printing the statement each in place here would be somewhat redundant.  Do you still want it?

Reply:
Ah true, missed that, this is fine as-is.